frontend/src/models/models.js

```diff
@@ -127,13 +137,25 @@ module.exports = app => app.component('models', {
         this.filter = eval(`(${this.searchText})`);
         this.filter = EJSON.stringify(this.filter);
         this.query.search = this.searchText;
-        this.$router.push({ path: this.$route.path, query: this.query })
       } else {
         this.filter = {};
         delete this.query.search;
-        this.$router.push({ path: this.$route.path, query: this.query });
+      }
+
+      const hashUrl = window.location.hash.replace(/^#/, '');
+      if (hashUrl.indexOf('?') === -1) {
+        window.history.pushState({}, '', window.location.pathname + '#' + hashUrl + '?search=' + this.query.search);
+      } else {
+        const searchParams = new URLSearchParams(
+          hashUrl.indexOf('?') === -1 ? '' : hashUrl.slice(hashUrl.indexOf('?') + 1)
+        );
+        const hashUrlWithoutSearchParams = hashUrl.slice(0, hashUrl.indexOf('?'));
+        
+        searchParams.set('search', this.query.search);
+        window.history.pushState({}, '', window.location.pathname + '#' + hashUrlWithoutSearchParams + '?' + searchParams);
       }
       await this.getDocuments();
+      this.applyQueryParams();
```

Comment:
I don't like having to re-apply query params here, that's unnecessary work. I think the better approach is to make a separate helper `this.loadMoreDocuments()` as a replacement for `this.getDocuments()` here. Because the reason why you need this `this.applyQueryParams()` call is that `this.getDocuments()` overwrites `this.filteredPaths`.

So make `loadMoreDocuments()` that takes the fetching logic from `getDocuments()`, but excludes the modifications to `schemaPaths`, `filteredPaths`, `selectedPaths`.